### PR TITLE
Replace terse for loop with map and for each

### PR DESCRIPTION
**Pull request by N8python**
I replaced the for loop, which was hard to understand at a glance, with a map and a forEach, which helps get the point across. In addition, I removed the terse ++ operator, replacing it with +=, and made both object declarations multiple lines. This code, while it sacrifices a few milliseconds of performance, allows an easier understanding of what the code does, and how it uses data.

### CODE REVIEW

File: src/vs/editor/contrib/bracketMatching/bracketMatching.ts
Comment (RDIL) [inline src/vs/editor/contrib/bracketMatching/bracketMatching.ts]: Issue - object is possibly `'null'`.  Needs fix.

File: src/vs/editor/contrib/bracketMatching/bracketMatching.ts
Comment (RDIL) [inline src/vs/editor/contrib/bracketMatching/bracketMatching.ts]: Issue - object is possibly `'null'`.  Needs fix.

**Review by RDIL — CHANGES_REQUESTED** [review]
The Azure DevOps build needs to pass before maintainers will merge this.  Here is what you need to keep in mind -

**Comment by alexandrudima** [comment]
Sorry, we don't accept PRs that only tweak code style...

**Pull request closed without merging**
